src/Effects/EffectMgr.cpp

```diff
@@ -177,29 +179,25 @@ static const rtosc::Ports local_ports = {
                 if (val > 0) {
                     eff->denominator = val;
                     int Pdelay, Pfreq;
-                    float freq;
-                    if(eff->numerator) {
+                    float freq, delay;
+                    if (eff->numerator&&eff->denominator) {
+                        eff->efx->speedfactor = (float)eff->denominator / (4.0f *(float)eff->numerator);
                         switch(eff->nefx) {
                         case 2: // Echo
-                            // invert:
-                            // delay = (Pdelay / 127.0f * 1.5f); //0 .. 1.5 sec
-                            Pdelay = (int)roundf((20320.0f / (float)eff->time->tempo) * 
-                                                 ((float)eff->numerator / (float)eff->denominator));
-                            if (eff->numerator&&eff->denominator)
-                                eff->seteffectparrt(2, Pdelay);
+                        case 10: // Reverse
+                            delay = 60.0f / ((float)eff->time->tempo * eff->efx->speedfactor);
```

Comment:
Should we add an assert that this will never divide by zero?

Reply:
i think it is impossible due to if (eff->numerator&&eff->denominator)

Reply:
i add an assertion for tempo